xml-3/src/test/java/com/baeldung/xml/XmlDocumentUnitTest.java

```diff
@@ -0,0 +1,53 @@
+package com.baeldung.xml;
+
+import org.junit.jupiter.api.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.xml.sax.InputSource;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.StringReader;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class XmlDocumentUnitTest {
+
+    @Test
+    public void givenXmlString_whenConvertToDocument_thenSuccess() throws Exception {
+        String xmlString = "<root><child>Example</child></root>";
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+
+        Document document = builder.parse(new InputSource(new StringReader(xmlString)));
+
+        assertNotNull(document);
+        assertEquals("root", document.getDocumentElement().getNodeName());
+
+        Element rootElement = document.getDocumentElement();
+        assertNotNull(rootElement.getElementsByTagName("child"));

```

Comment:
Since we are using `rootElement.getElementsByTagName("child")` 3 times here, let's extract that into a variable and update each test accordingly to use the variable.